### PR TITLE
Run some of Grist's browser tests against the desktop app

### PR DESCRIPTION
**Pull request by paulfitz**
The desktop app now gets exercised by the some of the same WebDriver browser tests Grist uses, so we have a chance of catching regressions here before they reach anyone.

A new runner, scripts/test-electron.js, launches the packaged Electron binary through chromedriver and drives it headless. It has two modes: a quick local pass over the desktop-only checks, and an --upstream mode that points Grist's own nbrowser suites at the desktop build instead of the regular app.

CI picks all this up: after the build it runs the quick tests first so simple breakage fails fast, then compiles the core suites (build:tests) and runs 13 of them against the desktop app. The 13 are just a random set that happen not to exercise multi-user behavior.

More useful will be tests of Desktop-specific behavior obviously! This is just a starting point.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,26 +23,29 @@
...
   "devDependencies": {
+    "chromedriver": "146",
     "fast-xml-parser": "^5.7.3",
     "resolve-tspaths": "^0.8.19",
     "typescript": "4.7.4"
   },
   "scripts": {
     "setup": "./scripts/setup.sh && cd core && yarn run install:python",
     "build": "cd core && yarn run build:prod && resolve-tspaths -p tsconfig-ext.json",
+    "build:tests": "cd core && ./node_modules/.bin/tsc --build test/tsconfig.json && resolve-tspaths -p test/tsconfig.json",
```
Comment (georgegevoian) [inline package.json:35]: I think just `tsc` should work here. Nitpicking though; this is fine too.

**Review by georgegevoian — APPROVED** [review]
Looking good. Great to have these running against desktop now.

Thanks Paul!

File: package.json (hunk 1 above)
Comment (paulfitz) [reply package.json:35]: Tried but gets muddled - gets the desktop tsc vs core tsc. Granted, it is weird using two builds like this, might need to revisit if it leads to problems, but for now some tests are better than none.
